live-usb-creator/README.md

```diff
@@ -73,17 +94,46 @@ In general there are 7 phases of the build process, all kicked off in sequence w
 
-```
+```bash
 sudo sh -c "stat --printf='%s' /var/tmp/tmp*/images/install.img | python -c 'import sys; print \"%0.1f%%\" % (float(int(sys.stdin.read())) / 1045090304.0 * 100.0)'"
 ```
 
 ## Hacking on this project
+
 Here are the useful files you may want to tweak.
 
 VM related stuff:
+
 * `Vagrantfile`: Provisions a VirtualBox VM in which the live system ISO will be built using the Vagrant tool.
 * `bootstrap.sh`: A shell script that runs at the very end of VM bringup. (Currently just mounts the CodeSafe ISO.)
-* `build.sh`: Manually run by the user to kick off the ISO build. Currently includes some patches for Lorax and Kickstart files applied before running the build proper with `livemedia-creator`.
+* `build.sh`: Manually run by the user to kick off the ISO build. Currently includes some patches for Lorax and Kickstart files applied before running the build proper with `livemedia-creator`. Two image types, development and release, can be built.
+  * Release build: `build.sh` or `build.sh release`.
+
+    A release image is to be used by cold storage operators who do not need
+    to have access to an interactive shell. For a release image to boot into
+    a root shell successfully, an SCSI/ATA hard drive needs to be available,
+    with a partition '/dev/sda1' used as persistent storage for the cold
+    wallet. The partition can be formatted in ext3 or vfat for example. The
+    device node `/dev/sda1` is required to give the operators a seamless
+    experience when booting up the system. A few other files, including the
+    HSM world and encrypted secrets, and compiled subzero code (in .jar and
+    signed .sar), are not checked in to the project repository, and need to
+    be added to their appropriate locations before building a release image
+    that will work out of the box.
+
+    * `data_app_subzero/` should contain `subzero-cli.jar`,
+    `subzero-signed.sar`, `subzero-userdata-signed`.
+    * `/dev/sda1` when mounted to `/hd`, should have the HSM and wallets files contained in `/hd/local` and `/hd/wallets`, respectively.
+
+  * Development build: `build.sh dev`.
+
+    A development image is to be used by subzero developers for HSM and cold
+    wallet development. A developer can boot into an interactive root shell
+    without performing the above mentioned steps. The HSM software is
+    automatically installed in the development image, but a developer needs
+    to manually set up persistent storage for the cold wallet, put .jar and
+    .sar files to appropriate filesystem locations, and execute them
```

Comment:
you shouldn't need to move any `.jar` or `.sar` files around.

Reply:
Good catch.